drivers/ieee802154/ieee802154_mcxw.c

```diff
@@ -600,16 +620,13 @@ static int mcxw_tx(const struct device *dev, enum ieee802154_tx_mode mode, struc
 	if (phy_status == gPhySuccess_c) {
 		mcxw_radio->tx_status = 0;
 		mcxw_radio->state = RADIO_STATE_TRANSMIT;
```

Comment:
move mcxw_radio->state = RADIO_STATE_TRANSMIT; before MAC_PD_SapHandler() to avoid a race condition with tx aborted event. Don't update the state if tx request (MAC_PD_SapHandler()) fails.
For rx, it's correct https://github.com/nxp-upstream/zephyr/blob/ecfda026b2b74885c68e1b279badce7b80fdc7b1/drivers/ieee802154/ieee802154_mcxw.c#L308